src/dependency-checks.js

```diff
@@ -0,0 +1,265 @@
+const debug = require('debug')('ember-m3');
+const pkg = require('../package.json');
+const semver = require('semver');
+const M3DataDeps = Object.keys(pkg.dependencies).filter(
+  k => k.indexOf('ember-data') !== -1 || k === 'ember-inflector'
+);
+
+const REQUIRED_DATA_PACKAGES = ['@ember-data/model', '@ember-data/store', 'ember-inflector'];
+const ALL_DATA_PACKAGES = [
+  '@ember-data/adapter',
+  '@ember-data/debug',
+  '@ember-data/model',
+  '@ember-data/record-data',
+  '@ember-data/serializer',
+  '@ember-data/store',
+];
+const minVersionForUsingOwnPackages = '3.14.0';
+
+class VersionChecker {
+  constructor(project) {
+    this._project = project;
+  }
+
+  *allAddons() {
```

Comment:
Fancy

Reply:
Not trying to be, but wanted to match ember-cli-version-checker's current API so we can PR this update (the existing implementation fails to catch most addons because until addons complete initialization the thing it iterates isn't populated)